pkg/cli/cmd/resourceprovider/create/create.go

```diff
@@ -98,6 +102,13 @@ func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
 	}
 	r.Workspace = workspace
 
+	if r.UCPClientFactory == nil {
+		err = r.setClientFactory(cmd.Context(), workspace)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Per offline discussing, we're moving this to `Run()` which will allow us to test `Validate()` more easily.